### PR TITLE
schema builds tables and inserts all platform combinations

### PR DESCRIPTION
**Pull request by DMcCallion**

### CODE REVIEW

File: live_pipeline/schema.sql [hunk 1]
```diff
@@ -0,0 +1,98 @@
+-- cleanup
+
+DROP TABLE IF EXISTS review;
+DROP TABLE IF EXISTS genre;
+DROP TABLE IF EXISTS game;
+DROP TABLE IF EXISTS developer;
+DROP TABLE IF EXISTS publisher;
+DROP TABLE IF EXISTS platform;
+
```
Comment (rhKashif) [inline live_pipeline/schema.sql:9]: General Question: Do we need to create a database for these tables to sit in? E.g.

```
DROP DATABASE IF EXISTS steampulse;
CREATE DATABASE steampulse;
\c steampulse;
```

File: live_pipeline/schema.sql [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+-- tables with no foreign key
+
+CREATE TABLE developer(
+    developer_id INT GENERATED ALWAYS AS IDENTITY,
+    developer_name TEXT NOT NULL,
```
Comment (rhKashif) [inline live_pipeline/schema.sql]: Would adding a `UNIQUE` constraint here be good? 

- We want to look into this table and pull out the ID of the developer if a matching developer is present.
- We want to add the developer to the table if not present.
- We don't want duplicate developers in this table. 

This should allow us to use `try-except` blocks when adding data which may be more easier.

File: live_pipeline/schema.sql [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    PRIMARY KEY (developer_id)
+
+);
+
+CREATE TABLE publisher(
+    publisher_id INT GENERATED ALWAYS AS IDENTITY,
+    publisher_name TEXT NOT NULL,
```
Comment (rhKashif) [inline live_pipeline/schema.sql]: Same question as above regarding `UNIQUE` constraint.

File: live_pipeline/schema.sql [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    PRIMARY KEY (platform_id)
+
+);
+
+-- game references the three above tables
+
+
+CREATE TABLE game(
+    game_id INT GENERATED ALWAYS AS IDENTITY,
+    app_id INT NOT NULL,
```
Comment (rhKashif) [inline live_pipeline/schema.sql]: Similar comment to adding a `UNIQUE` constraint

File: live_pipeline/schema.sql [hunk 5]
```diff
@@ -0,0 +1,98 @@
+-- cleanup
```
Comment (gawsalya) [inline live_pipeline/schema.sql:1]: Will there be a create database _database_name_ and /c _database_name_

File: live_pipeline/schema.sql [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+
+-- review and genre reference game
+
+-- default 0 might make sentiment analysis loading easier, but it's easy to take out if we don't use it
+
+CREATE TABLE review(
+    review_id INT GENERATED ALWAYS AS IDENTITY,
+    sentiment FLOAT NOT NULL DEFAULT 0,
+    review_text TEXT,
```
Comment (rhKashif) [inline live_pipeline/schema.sql]: Is there a reason for not having a `NOT NULL` constraint? 

- Do we want to add reviews with missing text?
- Are there reviews with missing text?

**Review by gawsalya — CHANGES_REQUESTED** [review]
just a comment about the database, the rest looks good and happy for merge

File: live_pipeline/schema.sql (hunk 5 above)
Comment (rhKashif) [inline live_pipeline/schema.sql:1]: General Comment: Change `INT` to `SMALLINT` where appropriate.

File: live_pipeline/schema.sql (hunk 6 above)
Comment (gawsalya) [reply live_pipeline/schema.sql]: I think this should get dropped in transform but yes agree would be wise to put in here as a double security measure

File: live_pipeline/schema.sql (hunk 1 above)
Comment (gawsalya) [reply live_pipeline/schema.sql:9]: If we do this then we wouldn't need to drop each table too!

**Pull request merged**